src/components/PostList/PostList.jsx

```diff
@@ -1,3 +1,43 @@
+import { PostInfo } from '../PostInfo';
+import postsFromServer from '../../api/posts.json';
+import commentsFromServer from '../../api/comments.json';
+import usersFromServer from '../../api/users.json';
+
+function getUserByPostId(id) {
+  const arr = [];
+
+  commentsFromServer.filter((elem) => {
+    if (elem.postId === id) {
+      arr.push(elem);
+    }
+
+    return elem;
+  });
+
+  return arr;
+}
+
+function getUserByUserId(userId) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const posts = postsFromServer.map(post => ({
+  ...post,
+  user: getUserByUserId(post.userId),
+}));
+
+const postsUpdate = posts.map((elem) => {
+  const valueElem = elem;
+
+  valueElem.comments = getUserByPostId(elem.id);
```

Comment:
do we really want to put our user into the comments array?
look at the `/api/comments.json`